lib/sablon/processor/content_type.rb

```diff
@@ -0,0 +1,28 @@
+module Sablon::Processor::ContentType
+  TYPES = {
+    png: 'image/png',
+    jpg: 'image/jpeg',
+    jpeg: 'image/jpeg',
+    gif: 'image/gif',
+    bmp: 'image/bmp'
+  }
+
+  def self.process(doc, properties, out)
+    TYPES.each do |extension, content_type|
+      unless extensions(doc).include?(extension.to_s)
```

Comment:
This will evaluate `extensions(doc)` once for every type in `TYPES`, when there’s no need — AFAICT, the result of the `#include?` test won’t change if this is only evaluated once before looping.